lui.el

```diff
@@ -1333,5 +1390,135 @@ which see for an explanation of the argument BUFFER-STRING."
                            front-sticky t))))
 
 
+;;;;;;;;;;;;;;;;;;
+;;; Navigation ;;;
+;;;;;;;;;;;;;;;;;;
+
+(defun lui-at-message-p ()
+  "Check if point is on a message."
+  (get-text-property (point) 'lui-message-id))
+
+(defun lui-beginning-of-message-p ()
+  "Check if point is at the beginning of a message."
+  (not (equal (get-text-property (point) 'lui-message-id)
+              (get-text-property (1- (point)) 'lui-message-id))))
+
+(defun lui-beginning-of-message ()
+  "Move point to the beginning of the message at point."
+  (goto-char (previous-single-property-change (point) 'lui-message-id)))
+
+(defun lui-forward-message ()
+  "Move point to the next message in the buffer and return point.
+If there is no next message, move to the end of the buffer instead."
+  (let ((current-id (get-text-property (point) 'lui-message-id))
+        (next-point
+         (next-single-property-change (point) 'lui-message-id)))
+    (if next-point
+        (let ((message-id
+               (get-text-property next-point 'lui-message-id)))
+          (goto-char next-point)
+          (when (or (not (or current-id message-id))
+                    (and current-id (not message-id))
+                    (and current-id message-id
+                         (= current-id message-id)))
+            (lui-forward-message)))
+      (goto-char (point-max)))
+    (point)))
```

Comment:
For readability, if the consequent clause if an `if` is very long, and the alternate is short, could you negate the condition and put the alternate first? (Alternatively, you can use `cond`, but I don't think that's necessary here)